metadata-ingestion/src/datahub/ingestion/transformer/extract_ownership_from_tags.py

```diff
@@ -22,13 +24,34 @@
 logger = logging.getLogger(__name__)
 
 
-class ExtractOwnersFromTagsConfig(TransformerSemanticsConfigModel):
-    tag_prefix: str
+class ExtractOwnersFromTagsConfig(ConfigModel):
+    tag_prefix: Optional[str]
+    tag_pattern: Optional[str]
     is_user: bool = True
+    owner_character_mapping: Optional[Dict[str, str]] = None
     email_domain: Optional[str] = None
+    extract_owner_type_from_tag_pattern: bool = False
     owner_type: str = "TECHNICAL_OWNER"
     owner_type_urn: Optional[str] = None
 
+    @root_validator(pre=True)
+    def raise_error_for_tag_prefix(cls, values: Dict) -> Dict:
```

Comment:
1. replace the validator with `pydantic_renamed_field`
2. remove the `tag_prefix` config
3. make `tag_pattern: str` instead of optional str